tests/Functional/Admin/BaseAdminTestCase.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineORMAdminBundle\Tests\Functional\Admin;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+abstract class BaseAdminTestCase extends WebTestCase
```

Comment:
WDYT about doing this class on src and marking it as experimental? This way we can reuse it on other bundles while not worrying for now about BC promises (we can remove experimental later once the class is stable enough) @VincentLanglet

Reply:
You mean doing this on the src class of SonataAdmin (instead of DoctrineORM) ?

It makes me think of this https://github.com/sonata-project/SonataAdminBundle/issues/7648

Reply:
Yes, tbh this class is quite simple. Not near as complex as the issue you mention. Also I think that proposal was more about adding some asserts for Sonata things (also a nice proposal). This one is like an abstraction around tipical things that you can test on an admin (Crud, filters, batch actions).

Reply:
See https://github.com/sonata-project/SonataCoreBundle/tree/3.x/src/CoreBundle/Test.

Reply:
Then, should we expose this as experimental? Or we can do it later on another PR if needed?

Reply:
Lets for now not expose this class.